recon/src/protocol.rs

```diff
@@ -44,15 +48,58 @@ use crate::{
 // Even a small limit will quickly mean that both peers have work to do.
 const PENDING_RANGES_LIMIT: usize = 20;
 
+// The max number of writes we'll batch up before flushing anything to disk.
+// As we descend the tree and find smaller ranges, this won't apply as we have to flush
+// before recomputing a range, but it will be used when we're processing large ranges we don't yet have.
+const INSERT_BATCH_SIZE: usize = 100;
+/// The maxium number of items we're willing to persist in memory for a given conversation until we find the
+/// necessary history to understand the event. Given a size of about 1KB/event, this is about 1MB per conversation.
+const MAX_PENDING_EVENTS: usize = 1000;
+
 type IM<K, H> = ReconMessage<InitiatorMessage<K, H>>;
 type RM<K, H> = ReconMessage<ResponderMessage<K, H>>;
 
+#[derive(Clone, Debug)]
+/// Parameters for the protocol that can be used to track the conversation or adjust internal behavior
+pub struct ProtocolConfig {
+    /// The max number of writes we'll batch up before flushing anything to disk.
+    /// As we descend the tree and find smaller ranges, this won't apply as we have to flush
+    /// before recomputing a range, but it will be used when we're processing large ranges we don't yet have.
+    pub insert_batch_size: usize,
+    /// The maximum number of items that can not be stored in memory before we begin to drop them.  
+    /// This is due to the event being dependent on another event we need to discover before it can be interpreted.
+    pub max_pending_items: usize,
+    #[allow(dead_code)]
+    /// The ID of the peer we're syncing with.
+    peer_id: PeerId,
```

Comment:
Added based on the discussion that this was needed for self anchoring work @smrz2001 @AaronGoldman

Reply:
huh, I'm surprised this is relevant to self anchoring?  Without seeing how this is going to be used though it's hard to reason about whether this actually needed or not.

Reply:
I believe the idea is to track which peer generated the event (at least in our eyes).. if it was us, we need to anchor it